src/hooks/use-axios.js

```diff
@@ -1,46 +1,61 @@
-import { useState, useEffect, useCallback } from 'react'
-import axios from 'axios'
+import { useEffect, useState } from 'react'
+
+import { useMutation, useQueryClient } from '@tanstack/react-query'
 
-const DELAY = 1 * 1000 // Artificial delay in ms
+import axios from 'axios'
 
-function useAxios(run = false, urlRun, bodyRun, methodRun, headersRun) {
+export default function useAxios(
+  run = false,
+  keys = ['key'],
+  urlRun,
+  bodyRun,
+  methodRun,
+  headersRun
+) {
   const [data, setData] = useState(null)
   const [error, setError] = useState('')
   const [loading, setLoading] = useState(false)
 
-  const callApiFn = useCallback(async (url, body, method, headers) => {
-    setLoading(true)
-    const options = body ? [url, body, { headers }] : [url, { headers }]
-    headers = headers || { 'content-type': 'application/json' }
-    method = method || (body ? 'post' : 'get')
+  const queryClient = useQueryClient()
 
-    try {
-      const res = await axios[method](...options)
-      const wait = (ms) => new Promise((res) => setTimeout(res, ms))
-      if (DELAY > 0) await wait(DELAY)
-      setData(res.data)
-      const { status } = res
-      if (status >= 400) {
-        setError(res)
-      } else {
-        setData(res.data)
-        setLoading(false)
+  const { mutateAsync: callApiFn } = useMutation({
+    mutationFn: async (args) => {
+      setLoading(true)
+      let { url, body, method, headers } = args
+      method = method || (body ? 'post' : 'get')
+      headers = headers || { 'content-type': 'application/json' }
+      const options = body ? [url, body, { headers }] : [url, { headers }]
+      try {
+        const res = await axios[method](...options)
+        if (res.status >= 400) return new Error(res.status)
         return res.data
+      } catch (err) {
+        return new Error(err)
       }
-    } catch (err) {
-      setError(err)
+    },
+    onSuccess: async (res) => {
+      setData(res.data)
+      setLoading(false)
+      queryClient.invalidateQueries(keys)
+    },
+    onError: async (error) => {
+      setError(error)
       setLoading(false)
-    }
-  }, [])
+    },
+    queryKey: [...keys.toString()],
```

Comment:
I don't think `queryKeys` exists on the `onMutation` hook params (see https://tanstack.com/query/v3/docs/framework/react/reference/useMutation). Do we need `keys` at all?

Reply:
You are right .. just removed any mention of key / keys ....